python_modules/dagster/dagster_tests/core_tests/definitions_tests/test_job.py

```diff
@@ -104,3 +106,19 @@ def my_op(y):
     result = my_job_with_input.execute_in_process(run_config={"inputs": {"x": {"value": 2}}})
     assert result.success
     assert result.output_for_node("my_op") == 2
+
+
+def test_job_post_process_config():
+    @op(config_schema={"foo": Field(StringSource)})
+    def the_op(context):
+        return context.op_config["foo"]
+
+    @graph
+    def basic():
+        the_op()
+
+    # Ensure that the env var not existing will not throw an error, since resolution happens in post-processing.
+    the_job = basic.to_job(config={"ops": {"the_op": {"config": {"foo": {"env": "SOME_ENV_VAR"}}}}})
```

Comment:
nit: assert the env var is in fact not set at this point (local test runs will inherit user env)